definitions/features/instance.rb

```diff
@@ -91,6 +91,24 @@ def upstream_install
     feature(:foreman_install) || feature(:katello_install)
   end
 
+  def current_version
+    version = if feature(:instance).downstream
+                feature(:instance).downstream.current_version
+              else
+                feature(:foreman_install).current_version
+              end
+
+    version.to_s[/^\d+\.\d+\.\d+/]
+  end
+
+  def target_version
+    if feature(:instance).downstream
+      Features::Satellite.new.target_version
```

Comment:
```suggestion
      feature(:satellite).target_version
```

Reply:
what about capsules?

Reply:
`Features::Satellite.new.target_version` is intentional - I want the target_version from this class and I do not want it to try to autodetect and fail this feature being present. I am trying to avoid defining target_version for Satellite in two places. 

The `target_version` being mocked was a good call out, that did not need to happen anymore and I updated the tests so that we can see it happen in Capsule and Satellite scenarios properly.

Reply:
Just to have it in writing what we talked about: this is not obvious from reading the code, especially as the `else` branch uses `feature()` again. I'll be OK if both branches use the same notation.